cert/selfsigned.go

```diff
@@ -100,6 +100,53 @@ func dnsNames(tlsExtraDomains []string) (string, []string) {
 	return host, dnsNames
 }
 
+// IsOutdated returns whether the given certificate is outdated w.r.t. the IPs
+// and domains given. The certificate is considered up to date if it was
+// created with _exactly_ the IPs and domains given.
+func IsOutdated(cert *x509.Certificate, tlsExtraIPs,
+	tlsExtraDomains []string) (bool, error) {
+
+	// Parse the slice of IP strings.
+	ips, err := ipAddresses(tlsExtraIPs)
+	if err != nil {
+		return false, err
+	}
+
+	// If the certificate has a different number of IP addresses, it is
+	// definitely out of date.
+	if len(ips) != len(cert.IPAddresses) {
+		return true, nil
+	}
+
+	// Go through each IP address, and check that they are equal. We expect
+	// the order to match exactly, as we assume the same tlsExtraIPs were
+	// used when the certificate was first created.
+	for i := range ips {
+		if !ips[i].Equal(cert.IPAddresses[i]) {
```

Comment:
related to above, imo order shouldn't matter here. plus we can reuse the sets created above to check. i think it'd be less obvious to users that reordering ip/dns/domains would cause their certs to be renewed